### PR TITLE
Approvals selector

### PR DESCRIPTION
**Pull request by bergarces**
## Explanation

This is one of the tickets aiming to refactor selectors to use the pending approvals. In this case, it replaces the one triggered with `wallet_watchAsset` method.

This should not affect the app functionality.

* Fixes https://github.com/MetaMask/MetaMask-planning/issues/451

## Manual Testing Steps

- Go to the [Test Dapp](https://metamask.github.io/test-dapp/)
- Scroll to the EIP 747 section
- Try to add an ERC20 Token, it should still work.

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [X] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/selectors/approvals.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { ApprovalRequest } from '@metamask/approval-controller';
+import { ApprovalType, Json } from '@metamask/controller-utils';
+
+type ApprovalsMetaMaskState = {
```
Comment (bergarces) [inline ui/selectors/approvals.ts:4]: This is a minimal type for the metamask state that represents what we need for this selector. There is no type that fully defines the state.

File: ui/selectors/approvals.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  metamask: {
+    pendingApprovals: ApprovalRequest<Record<string, Json> | null>[];
+  };
+};
+
+export function hasPendingApprovalsSelector(
+  state: ApprovalsMetaMaskState,
+  approvalType: ApprovalType,
+) {
+  const pendingApprovalRequests = Object.values(state.metamask.pendingApprovals)
```
Comment (bergarces) [inline ui/selectors/approvals.ts:14]: I was wondering if I had to check for undefined at some point here, but none of the references to this or many other state properties do across the code, so I'm assuming it's safe.

**Review by OGPoyraz — DISMISSED** [review]
LGTM

File: ui/selectors/approvals.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    .filter(({ type }) => type === approvalType)
+    .map(({ requestData }) => requestData);
```
Comment (OGPoyraz) [inline ui/selectors/approvals.ts:16]: Could we remove the map chaining here?

**Review by OGPoyraz — DISMISSED** [review]

**Review by OGPoyraz — DISMISSED** [review]

**Review by vinistevam — DISMISSED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Pull request merged**